### PR TITLE
For #1462: Lock implementation

### PR DESCRIPTION
**Pull request by paulodamaso**
For #1462: Lock implementation:
- Added implementation to `lock`, `unlock` and `isLocked`;
- Added IT cases testing these behaviors.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1464](https://github.com/jcabi/jcabi-github/issues/1464) is now in scope, role is `REV`

File: src/main/java/com/jcabi/github/RtIssue.java [hunk 1]
```diff
@@ -150,17 +156,42 @@ public void react(final Reaction reaction) {
...
+        final JsonStructure json = Json.createObjectBuilder()
+            .add("lock_reason", reason)
+            .build();
+        try{
+            this.request.method(Request.PUT).uri().path("/lock").back()
+                .body().set(json).back()
+                .fetch()
+                .as(RestResponse.class)
+                .assertStatus(HttpURLConnection.HTTP_NO_CONTENT);
+        } catch (IOException error){}
```
Comment (amihaiemil) [inline src/main/java/com/jcabi/github/RtIssue.java]: @paulodamaso why swallow the exception?

File: src/main/java/com/jcabi/github/RtIssue.java [hunk 2]
```diff
@@ -150,17 +156,42 @@ public void react(final Reaction reaction) {
...
 
     @Override
     public void unlock() {
-        throw new UnsupportedOperationException("unlock not implemented");
+        try{
+            this.request.method(Request.DELETE).uri().path("/lock").back()
+                .fetch()
+                .as(RestResponse.class)
+                .assertStatus(HttpURLConnection.HTTP_NO_CONTENT);
+        } catch (IOException error){}
```
Comment (amihaiemil) [inline src/main/java/com/jcabi/github/RtIssue.java]: @paulodamaso same here

**Comment by paulodamaso** [comment]
@amihaiemil ping

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Job `gh:jcabi/jcabi-github#1464` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#1464](https://github.com/jcabi/jcabi-github/issues/1464) is now out of scope

**Pull request merged**